### ISSUE TITLE
Roadside parking should ensure cars face correct direction

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#366]
Just noticed this while doing some testing (I have vehicles driving on left side of road as per UK driving)...

![wrong way](https://user-images.githubusercontent.com/1386719/59150493-88c77080-8a1c-11e9-9502-925b4ddb3c7f.png)

Both segments of the road shown above have roadside parking. From viewing orientation of the image above:

* On left segment, vehicles are parked facing the correct way (same direction as that side of road).
* On right segment, vehicles are parked facing wrong way (against traffic for that side of road).

I watched a car park on both segments, and noticed the issue when the car parking on right segment did a 180° turn while parking so it was facing the wrong way for traffic on that side of road while parked.

Can't remember how I built those segments (eg. might have built one of them from left to right, the other from right to left) but also have vague recollections that even when drawing long stretch of road, the game alternates segment direction to avoid texture 'banding' (or whatever it's called).

It would be nice if TM:PE could make cars face the correct direction when using roadside parking; specifically they should face same direction as traffic would travel on that side of the road.

**Comment by VictorPhilipp** [comment]
Is the behavior consistent such that given a segment cars always park with the same orientation?

**Comment by aubergine10** [comment]
Yes, it is consistent. It's most noticeable when there are multiple consecutive segments without junctions, as there you'll get the game swapping every other segment direction to avoid texture banding.

**Comment by VictorPhilipp** [comment]
I am unable to reproduce your problem locally. Is this a specific road you are using? Savegame?

**Comment by aubergine10** [comment]
The roads I'm using are the parking roads from this pack: https://steamcommunity.com/sharedfiles/filedetails/?id=1601040013

I'm not sure if it only occurs when roads are being drawn in different directions, or maybe it's something to do with using Move It to adjust node position and segment curvature, etc?

**Comment by aubergine10** [comment]
Another user mentioned similar issue with the same road pack, so it seems specific to these road assets.

> Plaxci Dzine™ 12 hours ago 
> Why do all the cars face one direction, even when the road is a two-way road. Is it a feature of the roads or a bug?

https://steamcommunity.com/sharedfiles/filedetails/?id=1601040013&tscn=1573454232

**Comment by aubergine10** [comment]
This issue is due to roads that specify `direction=both` on parking lanes. See https://github.com/CitiesSkylinesMods/TMPE/issues/516#issuecomment-587837970 for details.

**Comment by kvakvs** [comment]
<offtopic>In some countries it is allowed to park against the traffic flow. And in some it is a hefty fine.</offtopic>

**Comment by aubergine10** [comment]
In #516 there was discussion about allowing parking on other side of road if a) there's no more than one oncoming lane of traffic and b) the `m_canCrossLanes = true` is set for the road. But even there, the direction would still ideally need to be set properly to avoid weirdness in the vehicle and parking AI.
